File: 0704/catkin_ws/src/beginner_tutorials/src/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include <beginner_tutorials/testMsg.h>

void chatterCallback(const beginner_tutorials::testMsg &msg)
{
  ROS_INFO("I heard: [%s] int = [%d]", msg.str.c_str(), msg.integer);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "listener");
  ros::NodeHandle n;
  ros::Subscriber sub = n.subscribe("chatter", 1000, chatterCallback);
  ros::spin();
  return 0;
}

File: 0704/catkin_ws/src/beginner_tutorials/scripts/listener.py
#!/usr/bin/env python

import rospy
from beginner_tutorials.msg import testMsg

def callback(msg):
    rospy.loginfo("I heard {} integer={}".format(msg.str, msg.integer))

def listener():
    rospy.init_node('listener')
    rospy.Subscriber('chatter', testMsg, callback)
    rospy.spin()

if __name__ == '__main__':
    try:
        listener()
    except rospy.ROSInterruptException: 
        print("error")
        pass
